avatica/src/main/java/org/apache/calcite/avatica/AvaticaStatement.java

```diff
@@ -106,14 +106,44 @@ protected void executeInternal(String sql) throws SQLException {
     try {
       // In JDBC, maxRowCount = 0 means no limit; in prepare it means LIMIT 0
       final long maxRowCount1 = maxRowCount <= 0 ? -1 : maxRowCount;
-      Meta.ExecuteResult x =
-          connection.prepareAndExecuteInternal(this, sql, maxRowCount1);
+      while (true) {
+        // TODO limit this?
```

Comment:
Yeah, this was something I struggled making a decision about. On one hand, if there are transient failures (backend servers behind a proxy), we'd like to keep retrying. But, how do you decide how many is enough? I have to agree that keeping an upper bound and making it configurable is probably the safest approach though.